mucommander-core/src/main/java/com/mucommander/ui/terminal/TerminalSettingsProvider.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * This file is part of muCommander, http://www.mucommander.com
+ *
+ * muCommander is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * muCommander is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package com.mucommander.ui.terminal;
+
+import com.jediterm.terminal.ui.settings.DefaultSettingsProvider;
+import com.mucommander.commons.conf.ConfigurationEvent;
+import com.mucommander.commons.conf.ConfigurationListener;
+import com.mucommander.commons.runtime.OsFamily;
+import com.mucommander.conf.MuConfigurations;
+import com.mucommander.conf.MuPreference;
+import com.mucommander.conf.MuPreferences;
+
+/**
+ * A class providing configuration for Terminal - it is based on DefaultSettingsProvider
+ * and overrides certain settings based on Preferences.
+ * Currently, only 'altSendsEscape' driven by USE_OPTION_AS_META_KEY setting (https://github.com/mucommander/mucommander/issues/933)
+ */
+public class TerminalSettingsProvider extends DefaultSettingsProvider implements ConfigurationListener {
+
+    private boolean altSendsEscape;
+
+    public TerminalSettingsProvider() {
+        if (OsFamily.MAC_OS.isCurrent()) {
+            altSendsEscape = MuConfigurations.getPreferences().getVariable(
+                    MuPreference.USE_OPTION_AS_META_KEY,
+                    MuPreferences.DEFAULT_USE_OPTION_AS_META_KEY);
+            MuConfigurations.addPreferencesListener(this);
+        } else {
+            altSendsEscape = super.altSendsEscape();
+        }
+    }
+
+    @Override
+    public boolean altSendsEscape() {
+        return altSendsEscape;
+    }
+
+    @Override
+    public void configurationChanged(ConfigurationEvent event) {
+        if (MuPreference.USE_OPTION_AS_META_KEY.equals(
```

Comment:
how about checking if that's the configuration that has changed without iterating the values of MuPreference by
```suggestion
        if (MuPreferences.USE_OPTION_AS_META_KEY.equals(event.getVariable())) {
```

Reply:
I don't like to operate on strings if we can use enums.... I would rather say that MuPreferences should be refactored to be enums, too.

Reply:
ok, in my opinion it's alright to use strings now that we can use them in switch-case but that's a matter of taste
I don't mind having an extra step of getting the enum from the label here but we can do it more efficiently - I'll post a separate cleanup